File: app/src/main/java/com/zqb/mvvmkotlin/widgets/Loading.java
package com.zqb.mvvmkotlin.widgets;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.zqb.mvvmkotlin.R;

public class Loading extends Dialog {
    public Loading(@NonNull Context context) {
        this(context, R.style.BaseDialog);
    }

    public Loading(@NonNull Context context, int themeResId) {
        super(context, themeResId);
    }

    protected Loading(@NonNull Context context, boolean cancelable,  @Nullable OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_load);
        setCanceledOnTouchOutside(false);
    }
}

File: app/src/main/java/com/zqb/mvvmkotlin/widgets/Loading.kt
package com.zqb.mvvmkotlin.widgets

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import com.zqb.mvvmkotlin.R

class Loading @JvmOverloads constructor(context: Context, themeResId: Int = R.style.BaseDialog) :
    Dialog(context, themeResId) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_load)
        setCanceledOnTouchOutside(false)
    }
}
